samples/wifi/scan/src/main.c

```diff
@@ -47,6 +47,316 @@ static struct net_mgmt_event_callback wifi_shell_mgmt_cb;
 
 K_SEM_DEFINE(scan_sem, 0, 1);
 
+#ifdef CONFIG_WIFI_SCAN_CHAN
+#include <ctype.h>
+#define WIFI_MAX_CHAN_STR_LEN 4
+#define WIFI_MAX_BAND_STR_LEN 3
+
+static enum wifi_frequency_bands wifi_map_band_str_to_idx(char *band_str)
+{
+	enum wifi_frequency_bands band = WIFI_FREQ_BAND_UNKNOWN;
+
+	if (!strcmp(band_str, "2")) {
+		band = WIFI_FREQ_BAND_2_4_GHZ;
+	} else if (!strcmp(band_str, "5")) {
+		band = WIFI_FREQ_BAND_5_GHZ;
+	} else if (!strcmp(band_str, "6")) {
+		band = WIFI_FREQ_BAND_6_GHZ;
+	}
+
+	return band;
+}
+
+static const uint16_t valid_5g_chans_20mhz[] = {32, 36, 40, 44, 48, 52, 56, 60, 64, 68, 96, 100,
+	104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 144, 149, 153, 157, 159, 161,
+	163, 165, 167, 169, 171, 173, 175, 177};
+
+static bool wifi_validate_chan_2g(uint16_t chan)
+{
+	if ((chan >= 1) && (chan <= 14)) {
+		return true;
+	}
+
+	return false;
+}
+
+
+static bool wifi_validate_chan_5g(uint16_t chan)
+{
+	uint16_t i;
+
+	for (i = 0; i < ARRAY_SIZE(valid_5g_chans_20mhz); i++) {
+		if (chan == valid_5g_chans_20mhz[i]) {
+			return true;
+		}
+	}
+	return false;
+}
```

Comment:
Don't duplicate the code from wifi_utils.c, either use those functions directly or keep it simple by removing all validations and let wifi_mgmt take care of it.